### PR TITLE
approach to get a package management page

### PR DESCRIPTION
**Pull request by Edru2**
This is an approach to solve the issue https://github.com/Mudlet/mudlet-package-repo/issues/21
~~It definitely not finished.~~
finished to my extend (definitely needs better styling)

Changing package details/updating package file and deleting a package should work.
Deleting won't delete the package file but just the package entry in the database.

~~Didn't change the button text yet (also didn't use the translation module yet) just wanted to see if it's functional.~~

### CODE REVIEW

**Comment by vadi2** [comment]
LuaRocks is broken so I can't test this. Can you add screenshots?

**Comment by Edru2** [comment]
Sorry for my delay. (unrelated to this) I managed to break my OS and had to reinstall everything.
About screenshot,
like I mentioned before there is not much visible going on right now (didn't even adjust button and success text) it looks more or less like the upload package one.
But just to have an idea: 
modify package

**Comment by vadi2** [comment]
This makes sense, only question I have - could both be the same page, because they share so much in common?

**Comment by Edru2** [comment]
I agree with putting everything on one page, I thought of that before myself. I'm just not sure if I'm able to 😅. Have to look at this again.

**Comment by vadi2** [comment]
If it makes things too complicated then duplication is fine!

File: views/changepackage.etlua [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      <% for k, v in pairs(myPackages) do %>
+          <a type = "button" onclick="fillValues('<%= v.name %>','<%= v.version %>','<%= v.description %>','<%= v.game %>','<%=v.dependencies %>')"> <%= v.name %></a>
+      <% end %>
+    </div>
+    </div>
+    <input type="text" name="package" id="package" style="display:none;"/>
+  </div>
+  <div class="form-group">
+    <label for="choosePackageFile">
+      <%= i18n("upload_button_header") %>
```
Comment (vadi2) [inline views/changepackage.etlua]: The upload buttons are duplicated, accident?

**Review by vadi2 — COMMENTED** [review]
Uploading failed with

`Unable to change working directory to 'data' No such file or directory `

Not sure using disk storage is the best way - BLOB storage with PostgreSQL might be the way to go, also so we can handle multiple versions of a package later on.

File: views/changepackage.etlua (hunk 1 above)
Comment (Edru2) [reply views/changepackage.etlua]: No, just not changed the text yet.
One button is to choose the package and the other to upload.

Comment (Edru2) [reply views/changepackage.etlua]: > Not sure using disk storage is the best way - BLOB storage with PostgreSQL might be the way to go, also so we can handle multiple versions of a package later on.

Sounds good.
That everything is saved in the data directory wasn't introduced by this PR, I can have a look how to change this but I think it's beyond my abilities.

Comment (vadi2) [reply views/changepackage.etlua]: My bad, OK, let's keep it that way for now then. I thought it was added.

File: static/i18n/en.lua [hunk 2]
```diff
@@ -1,6 +1,8 @@
 return {
   en = {
     upload_success = "Package %s uploaded successfully",
+    change_success = "Package %s modified succesfully",
+    delete_success = "Package %s deleted succesfully",
```
Comment (vadi2) [inline static/i18n/en.lua]:
```suggestion
    change_success = "Package %s modified successfully",
    delete_success = "Package %s deleted successfully",
```

File: controllers/changepackage.lua [hunk 3]
```diff
@@ -73,15 +73,23 @@ return {
     assert_error(self.session.name, self.i18n("err_not_logged_in"))
     assert_error(self.session.verified, self.i18n("err_email_not_verified"))
     local user = Users:get_user(self.session.name)
-    self.myPackages = Packages:select("* where user_id = "..user.id)
+    if not user.admin then
+      self.myPackages = Packages:select("* where user_id = "..user.id)
```
Comment (vadi2) [inline controllers/changepackage.lua]: Don't concatenate values like this - it's how SQL injections happen :wink: check out https://leafo.net/lapis/reference/database.html#making-a-query on how to do them instead.

Comment (Edru2) [reply controllers/changepackage.lua]: I put it into models and it's not a concatenation anymore :)

Comment (vadi2) [reply controllers/changepackage.lua]: Very nice :)

File: static/i18n/en.lua [hunk 4]
```diff
@@ -30,8 +32,13 @@ return {
     error_header = "ERROR",
     navigation_title = "Navigation links",
     upload_button_header = "Select package to upload",
+    update_button_header = "Update your package file",
+    change_button_header = "Select your package",
     choose_file = "Choose Package File",
+    choose_package = "Choose Package to modify",
     upload_package = "Upload Package",
+    change_package = "Modifiy Package",
```
Comment (vadi2) [inline static/i18n/en.lua:40]:
```suggestion
    change_package = "Modify Package",
```

**Review by vadi2 — APPROVED** [review]
Looks good, let's resolve any issues once it's merged

**Pull request merged**